v0/destinations/mailmodo/transform.js

```diff
@@ -0,0 +1,266 @@
+
+  const response = defaultRequestConfig();
+  if (resp) {
+    response.endpoint = `${BASE_URL}${endpoint}`;
+    response.headers = {
+      mmApiKey: `${apiKey}`,
+      "Content-Type": "application/json"
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = removeUndefinedAndNullValues(resp);
+  }
+  return response;
+};
+
+const identifyResponseBuilder = (message, { Config }) => {
+  const { apiKey } = Config;
+  let { listName } = Config;
+
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategory.IDENTIFY.name]
+  );
+
+  const { address1, address2 } = deduceAddressFields(message);
+  if (address1) payload.data.address1 = address1;
+  if (address2) payload.data.address2 = address2;
+
+  if (typeof listName === "string" && listName.trim().length === 0) {
+    listName = "Rudderstack";
+  }
+
+  const valuesArray = [payload];
+  const resp = { values: valuesArray, listName };
+
+  const { endpoint } = ConfigCategory.IDENTIFY;
+
+  const responseConfgs = {
+    resp,
+    apiKey,
+    endpoint
+  };
+  return responseBuilder(responseConfgs);
+};
+
+const trackResponseBuilder = (message, { Config }) => {
+  const { apiKey } = Config;
+  const resp = constructPayload(
+    message,
+    mappingConfig[ConfigCategory.TRACK.name]
+  );
+
+  const { endpoint } = ConfigCategory.TRACK;
+
+  const responseConfgs = {
+    resp,
+    apiKey,
+    endpoint
+  };
+  return responseBuilder(responseConfgs);
+};
+
+const processEvent = (message, destination) => {
+  if (!destination.Config.apiKey) {
+    throw new CustomError("API Key is not present. Aborting message.", 400);
+  }
+  if (!message.type) {
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+  const messageType = message.type.toLowerCase();
+
+  let response;
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response = identifyResponseBuilder(message, destination);
+      break;
+    case EventType.TRACK:
+      response = trackResponseBuilder(message, destination);
+      break;
+    default:
+      throw new CustomError("Message type not supported", 400);
+  }
+  return response;
+};
+
+const process = event => {
+  return processEvent(event.message, event.destination);
+};
+
+function batchEvents(eventsChunk) {
+  const batchedResponseList = [];
+
+  const arrayChunks = _.chunk(eventsChunk, IDENTIFY_MAX_BATCH_SIZE);
+
+  // list of chunks [ [..], [..] ]
+  arrayChunks.forEach(chunk => {
+    const metadatas = [];
+    const values = [];
+
+    // extracting destination
+    // from the first event in a batch
+    const { destination } = chunk[0];
+    const { apiKey } = destination.Config;
+    let { listName } = destination.Config;
+
+    // listName will be "Rudderstack", if it is not provided
+    if (typeof listName === "string" && listName.trim().length === 0) {
+      listName = "Rudderstack";
+    }
```

Comment:
This is not required as we are already setting `listName` in `process` function.

Reply:
Removed